go-controller/pkg/ovn/address_set/address_set.go

```diff
@@ -74,6 +77,51 @@ func (asf *ovnAddressSetFactory) NewAddressSet(name string, ips []net.IP) (Addre
 	return res, nil
 }
 
+// EnsureAddressSet ensures the address_set with the given name exists and if it does not creates an empty addressSet
+func (asf *ovnAddressSetFactory) EnsureAddressSet(name string) error {
+	hashedAddressSetNames := []string{}
+	ip4ASName, ip6ASName := MakeAddressSetName(name)
+	if config.IPv4Mode {
+		hashedAddressSetNames = append(hashedAddressSetNames, ip4ASName)
+	}
+	if config.IPv6Mode {
+		hashedAddressSetNames = append(hashedAddressSetNames, ip6ASName)
+	}
+	for _, hashedAddressSetName := range hashedAddressSetNames {
+		uuid, stderr, err := util.RunOVNNbctl(
+			"--data=bare",
+			"--no-heading",
+			"--columns=_uuid",
+			"find",
+			"address_set",
+			"name="+hashedAddressSetName)
+		if err != nil {
+			return fmt.Errorf("find failed to get address set %q, stderr: %q (%v)",
+				name, stderr, err)
+		}
+		if uuid != "" {
+			// address_set already exists
+			continue
+		} else {
```

Comment:
@JacobTanenbaum I would just:

```
if uuid != "" {
    ...
}

<create it>
```

to remove one level of indent on the creation call.

Reply:
done